integration-tests/subscriptions.test.ts

```diff
@@ -39,4 +46,39 @@ if (SUBSCRIPTIONS_ENABLED === 'true') {
             console.log(x);
         });
     });
+
+    let client: AxiosInstance;
+    describe('test subscription creation and deletion', () => {
+        beforeAll(async () => {
+            client = await getFhirClient({ tenant: MULTI_TENANCY_ENABLED ? 'tenant1' : undefined });
+        });
+
+        test('creation of almost expiring subscription should be deleted by reaper', async () => {
+            // OPERATE
+            const subscriptionResource = {
+                resourceType: 'Subscription',
+                status: 'requested',
+                // get a time 10 seconds (10000 ms) in the future
+                end: new Date(new Date().getTime() + 10000).toISOString(),
+                reason: 'Monitor Patients for Organization 123',
+                criteria: 'Patient?name=Smith',
+                channel: {
+                    type: 'rest-hook',
+                    endpoint: 'https://customer-endpoint.com',
+                    payload: 'application/fhir+json',
+                    header: ['Authorization: Bearer secret-token-abc-123'],
+                },
+            };
+            const postSubResult = await client.post('Subscription', subscriptionResource);
+            expect(postSubResult.status).toEqual(201); // ensure that the sub resource is created
+            const subResourceId = postSubResult.data.id;
+            // wait 4 min until reaper has had a chance to run
+            // (waiting 5 min would hit the timeout value above)
+            // it may be better to do a loop checking every minute or so or increasing the timeout
+            await waitForMs(4 * 60 * 1000);
+            await expect(client.get(`Subscription/${subResourceId}`)).rejects.toMatchObject({
+                response: { status: 404 },
+            });
```

Comment:
let's check every 30 seconds instead of waiting for a flat 4 minutes (also, 4 minutes is not enough to guarantee that the reaper runs)

you can use `waitForExpect`, a library we already use that does the checks in a loop. 

Here's an example: https://github.com/awslabs/fhir-works-on-aws-deployment/blob/c9a2c78c560a67e69277df572e8d1c9f1183aa88/integration-tests/implementationGuides.test.ts#L249-L263

something like: 
```ts
await waitForExpect(
            () => {
                console.log('Checking if Subscription/111 has already been deleted');
                expect(client.get(`Subscription/111`)).rejects.toMatchObject({
                    response: { status: 404 },
                });
            },
            360_000,
            30_000,
        );
```

Reply:
actually, it was missing async await before
```ts
await waitForExpect(
            async () => {
                console.log('Checking if Subscription/111 has already been deleted');
                await expect(client.get(`Subscription/111`)).rejects.toMatchObject({
                    response: { status: 404 },
                });
            },
            360_000,
            30_000,
        );
```